executor/load_data.go

```diff
@@ -249,12 +278,51 @@ type LoadDataWorker struct {
 
 // NewLoadDataWorker creates a new LoadDataWorker that is ready to work.
 func NewLoadDataWorker(
-	sctx sessionctx.Context,
+	userSctx sessionctx.Context,
 	plan *plannercore.LoadData,
 	tbl table.Table,
 	getSysSessionFn func() (sessionctx.Context, error),
 	putSysSessionFn func(context.Context, sessionctx.Context),
-) (*LoadDataWorker, error) {
+) (w *LoadDataWorker, err error) {
+	ownSession := false
+	defer func() {
+		if err != nil && ownSession {
+			ctx := context.Background()
+			ctx = kv.WithInternalSourceType(ctx, kv.InternalLoadData)
+			putSysSessionFn(ctx, userSctx)
+		}
+	}()
+
+	controller := importer.NewLoadDataController(plan, tbl)
+	if err = controller.Init(userSctx, plan.Options); err != nil {
+		return nil, err
+	}
+
+	sctx := userSctx
+	if controller.Detached {
+		if plan.FileLocRef == ast.FileLocClient {
+			return nil, exeerrors.ErrLoadDataCantDetachWithLocal
+		}
+		sysSession, err2 := getSysSessionFn()
+		if err2 != nil {
+			return nil, err2
+		}
+		ownSession = true
+
+		err = ResetContextOfStmt(sysSession, &ast.LoadDataStmt{})
+		if err != nil {
+			return nil, err
+		}
+		// copy the related variables to the new session
+		// I have no confident that all needed variables are copied :(
+		from := userSctx.GetSessionVars()
+		to := sysSession.GetSessionVars()
+		to.User = from.User
+		to.CurrentDB = from.CurrentDB
+		to.SQLMode = from.SQLMode
```

Comment:
Maybe you could ref https://github.com/pingcap/tidb/blob/fff7fb9fecb1149df5982c302197227901ed1693/ddl/backfilling.go#L805 to determine which variable are useful.

Not sure timezone is useful for `load_data` or not.

Reply:
I added a simple test for related variables https://github.com/pingcap/tidb/pull/42159/commits/0b82a0aa2fe5ef7cd54a81830501461b366c5a43